packages/manager/apps/dedicated/client/app/dedicatedCloud/datacenter/backup/backup.class.js

```diff
@@ -0,0 +1,81 @@
+import {
+  BACKUP_STATE_DISABLED,
+  BACKUP_STATE_REMOVING,
+  BACKUP_STATE_ENABLED,
+  BACKUP_OFFER_LEGACY,
+  BACKUP_OFFER_ADVANCED,
+  BACKUP_OFFER_PREMIUM,
+} from './backup.constants';
+
+export default class Backup {
+  constructor({
+    backupDurationInReport,
+    backupOffer,
+    backupSizeInReport,
+    diskSizeInReport,
+    encryption,
+    fullDayInReport,
+    hostname,
+    mailAddress,
+    replicationZone,
+    restorePointInReport,
+    scheduleHour,
+    state,
+    vmwareVmId,
+  }) {
+    Object.assign(this, {
+      backupDurationInReport,
+      backupOffer,
+      backupSizeInReport,
+      diskSizeInReport,
+      encryption,
+      fullDayInReport,
+      hostname,
+      mailAddress,
+      replicationZone,
+      restorePointInReport,
+      scheduleHour,
+      state,
+      vmwareVmId,
+    });
+
+    this.backupSizeInReport = backupSizeInReport || false;
+    this.fullDayInReport = fullDayInReport || false;
+    this.restorePointInReport = restorePointInReport || false;
+    this.diskSizeInReport = diskSizeInReport || false;
+    this.encryption = encryption || false;
+    this.backupDurationInReport = backupDurationInReport || false;
+  }
+
+  get sendReportToCustomerEmail() {
+    return this.mailAddress !== null;
+  }
+
+  isInactive() {
+    return this.state === BACKUP_STATE_DISABLED;
+  }
+
+  isActive() {
+    return this.state === BACKUP_STATE_ENABLED;
+  }
+
+  isRemoving() {
+    return this.state === BACKUP_STATE_REMOVING;
+  }
+
+  isLegacy() {
+    return this.backupOffer === BACKUP_OFFER_LEGACY;
+  }
+
+  isAdvanced() {
+    return this.backupOffer === BACKUP_OFFER_ADVANCED;
+  }
+
+  isPremium() {
+    return this.backupOffer === BACKUP_OFFER_PREMIUM;
+  }
+
+  setState(state) {
+    this.state = state;
```

Comment:
Why not directly use the setter ?

Reply:
The method was used earlier, since we did not know what properties the class had. But now, since we are destructuring in the constructor, we can set the property value directly and even a setter would not be needed. Please let me know your thought